git/commit.go

```diff
@@ -1,12 +1,20 @@
 package git
 
 import (
+	"regexp"
+
 	"github.com/gogs/git-module"
 )
 
 // ZeroHash is the zero hash.
 var ZeroHash Hash = git.EmptyID
 
+// IsZeroHash returns whether the hash is a zero hash.
+func IsZeroHash(h Hash) bool {
+	pattern := regexp.MustCompile(`^[0]+$`)
+	return pattern.MatchString(h.String())
```

Comment:
maybe `return h.String() == ZeroHash.String()` works?

Reply:
Trying to be future proof as Git now supports sha256 hashes https://git-scm.com/docs/hash-function-transition/